File: lambda/subscribeToLive.js
'use strict';
var AWS = require("aws-sdk");
var https = require('https');

const secretsManager = new AWS.SecretsManager({region: "us-east-1"});

// https://gist.github.com/rxgx/7e1b24de5936ff1b2b815a3d9cc3897a

exports.handler = async (event, context, callback) => {

    event.Records.forEach((record) => {
        console.log('Stream record: ', JSON.stringify(record, null, 2));
        if(record.eventName == "INSERT"){

          const data = await secretsManager.getSecretValue({SecretId: "twitch.client_id",}).promise();
          const client_id = JSON.parse(data.SecretString).twitch.client_id;

          const user_id = record.dynamodb.NewImage.S.id;  // Get this from record ...
          const access_token = record.dynamodb.NewImage.S.access_token;  // Get this from record ...

          const data = JSON.stringify({
              "hub.callback": "http://460ec3f71132.ngrok.io/webhook",  // Should the url of our webhook
              "hub.mode": "subscribe",
              "hub.topic": `https://api.twitch.tv/helix/streams?user_id=${user_id}`,
              "hub.lease_seconds": 432000
            })

            const options = {
              hostname: 'api.twitch.tv',
              port: 443,
              path: '/helix/webhooks/hub',
              method: 'POST',
              headers: {
                'Content-Type': 'application/json',
                'Content-Length': data.length,
                'Authorization': `Bearer ${access_token}`,
                'Client-ID': client_id
              }
            }

            const req = https.request(options, (res) => {
              console.log(`statusCode: ${res.statusCode}`)
              callback(null, "Got response: " + res.statusCode);
              // res.setEncoding('utf8');
              // res.on('data', (d) => { process.stdout.write(d) })
            })

            req.on('error', (error) => {
              console.error(error)
              callback(new Error(error));
            })

            req.write(data)
            req.end()
        }
        else callback(null, `Successfully processed ${event.Records.length} records.`);
    });
};

File: lambda/subscribeToLive.py
import json
from botocore.vendored import requests
import boto3


def lambda_handler(event, context):
    for record in event["Records"]:
        if record["eventName"] == "INSERT":
            print("Current record: {}".format(record))

            session = boto3.session.Session()
            secret_manager = session.client(service_name='secretsmanager', region_name="us-east-1")
            client_id = secret_manager.get_secret_value(SecretId="twitch.client_id")
            client_id = json.loads(client_id["SecretString"])["twitch"]["client_id"]

            user_id = record["dynamodb"]["NewImage"]["id"]["S"]  # Get this from record ...
            access_token = record["dynamodb"]["NewImage"]["access_token"]["S"]  # Get this from record ...

            data = {
                "hub.callback": "http://twitc-ecsal-it3rj1p6kk2r-780238624.us-east-1.elb.amazonaws.com/api/v1/webhook/{}".format(user_id),  # Should the url of our webhook
                "hub.mode": "subscribe",
                "hub.topic": "https://api.twitch.tv/helix/streams?user_id={}".format(user_id),
                "hub.lease_seconds": 432000
            }
            print(data)

            response = requests.post("https://api.twitch.tv/helix/webhooks/hub", data=data, headers={
                "Authorization": "Bearer {}".format(access_token),
                "Client-ID": client_id
            })

            return {'status_code': response.status_code, 'text': response.text}
    return {'status_code': 200, 'text': "Nothing to update"}
